### PR TITLE
test: migrate some files to snapbox

### PR DESCRIPTION
**Pull request by eth3lbert**
### What does this PR try to resolve?

Part of https://github.com/rust-lang/cargo/issues/14039.

Migrate following to snapbox:

- `tests/testsuite/local_registry.rs`
- `tests/testsuite/locate_project.rs`
- `tests/testsuite/lockfile_compat.rs`
- `tests/testsuite/login.rs`
- `tests/testsuite/logout.rs`

### CODE REVIEW

File: tests/testsuite/local_registry.rs [hunk 1]
```diff
@@ -241,17 +240,16 @@ fn multiple_names() {
...
-[CHECKING] [..]
-[CHECKING] foo v0.0.1 ([CWD])
-[FINISHED] [..]
-",
-        )
+[UNPACKING] bar v0.0.1 (registry `[ROOT]/registry`)
+[UNPACKING] baz v0.1.0 (registry `[ROOT]/registry`)
+[CHECKING] ba[..] v0.[..]
+[CHECKING] ba[..] v0.[..]
+[CHECKING] foo v0.0.1 ([ROOT]/foo)
```
Comment (epage) [inline tests/testsuite/local_registry.rs:249]: How come we are doing these globs?  If its unordered should we instead set that property?

Comment (eth3lbert) [reply tests/testsuite/local_registry.rs:249]: Yes, to address the ordering issue, I'll use `.unordered()` instead!

Comment (eth3lbert) [reply tests/testsuite/local_registry.rs]: Updated!

**Pull request merged**